modular_skyrat/modules/modular_implants/code/nif_presistance.dm

```diff
@@ -0,0 +1,98 @@
+/// How much damage is done to the NIF if the user ends the round with it uninstalled?
+#define LOSS_WITH_NIF_UNINSTALLED 25
+
+///Saves the NIF data for everyone inside of the server.
+/datum/controller/subsystem/persistence/proc/save_nifs()
+	for(var/player in GLOB.joined_player_list)
+		var/mob/living/carbon/human/ending_human = get_mob_by_ckey(player)
+		if(!istype(ending_human) || !ending_human.mind?.original_character_slot_index)
+			continue
+
+		var/mob/living/carbon/human/original_human = ending_human.mind.original_character.resolve()
+
+		if(!original_human)
+			continue
+
+		original_human.save_nif_data()
+
+///Saves the NIF data for a individual user.
+/mob/living/carbon/human/proc/save_nif_data()
+	if(!ckey || !mind?.original_character_slot_index)
+		return
+
+	var/path = "data/player_saves/[ckey[1]]/[ckey]/nif.sav"
```

Comment:
Needs to hook into pref files for easy per character pref loading, please don't use sav files, cause if they even slightly corrupt, there's no easy or quick way to clear out broken data without deleting the entire thing.

Feel free to poke me if you need pointers on what to do for adding to prefs.